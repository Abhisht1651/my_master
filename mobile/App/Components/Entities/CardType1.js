// @flow
// import React, { useState } from 'react'
import React, { useState, useEffect, useRef } from 'react'

import { View, Text, Image } from 'react-native'
import PropTypes from 'prop-types'
import { TouchableOpacity } from 'react-native-gesture-handler'
import {withApollo} from 'react-apollo';

// Styles
import styles from '../Styles/EntitesStyles/CardType1Styles'
import images from '../../Themes/Images'
import { DefaultButton } from '../Buttons/DefaultButton'
import { X4SButton } from '../Buttons/X4SButton'
import {getEntityDetails,getRelatedEntities,getMultimediaDetails} from '../../Containers/StoryLine/my_queries';


// type Props = {
//   title: string,
//   count: number,
//   image1: Image,
//   onPressFollow: PropTypes.func
// }

const CardType1 = (props) => {
  const client=props.client;

  if (!client){
    console.log("CLient not working")
  }
  const { title, count,profileIcons,onPressFollow,is_following } = props


  // const onPressFollow= () => {}
  const [pressStatus, setStatus] = useState(is_following)
  const changeStatus = () => {
    onPressFollow(!pressStatus)
    setStatus(!pressStatus)
  }
 

  const profileIconElement = profileIcons.map((profileIcon, index) => (
    <Image source={{uri:profileIcon.toString()}} key={index} style={styles.profileIcon} />
  ))

  return (
    <View style={styles.cardBox}>
      <View style={styles.profileIconBox}>{profileIconElement}</View>
      <View style={styles.textOutline}>
        <View>
          <Text style={styles.textNameFont}>{title}</Text>
        </View>
        <View>
          <Text style={styles.textDetailsFont}>{`${count}K Following`}</Text>
        </View>
      </View>

      <View style={styles.followbuttonBox}>
        <TouchableOpacity onPress={changeStatus}>
          {!pressStatus ? (
            <DefaultButton outline title="Follow" onPress={onPressFollow} />
          ) : (
            <X4SButton disabled title="Following" onPress={onPressFollow} />
          )}
        </TouchableOpacity>
      </View>
    </View>
  )
}

// CardType1.defaultProps = {
//   title: 'Default',
//   count: 0,
//   image1: images.EntitiesImage0,
//   onPressFollow: () => {}
// }

export default withApollo(CardType1)

