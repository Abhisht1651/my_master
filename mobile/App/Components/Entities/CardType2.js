import React from 'react'
import { View, Text, Image } from 'react-native'
import PropTypes from 'prop-types'

// Styles
import styles from '../Styles/EntitesStyles/CardType2Styles'
import images from '../../Themes/Images'

const CardType2 = ({ type, title, count, profileIcons }) => {
  const profileIconElement = profileIcons.map((profileIcon, index) => (
    <Image
      source={profileIcon}
      key={index}
      style={type === 1 ? styles.profileIcon : styles.profileIconRound}
    />
  ))
  return (
    <View style={styles.cardBox}>
      <View style={styles.profileIconBox}>{profileIconElement}</View>

      <View style={styles.textOutline}>
        <View style={styles.textcenter}>
          <Text style={styles.textNameFont}>{title}</Text>
        </View>
        <View>
          <Text style={styles.textDetailsFont}>{`${count} friends`}</Text>
        </View>
        <View>
          {type === 1 ? (
            <Text style={styles.textDetailsFont}>follow</Text>
          ) : (
            <Text style={styles.textDetailsFont}>in common</Text>
          )}
        </View>
      </View>
    </View>
  )
}

CardType2.propTypes = {
  type: PropTypes.number,
  title: PropTypes.string,
  count: PropTypes.number,
  profileIcons: PropTypes.array
}

CardType2.defaultProps = {
  type: 1,
  title: 'Default',
  count: 0,
  profileIcons: [images.EntitiesImage0]
}

export default CardType2
