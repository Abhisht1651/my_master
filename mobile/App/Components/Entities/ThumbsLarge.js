import React from 'react'
import { View, Text, Image } from 'react-native'
import PropTypes from 'prop-types'

// Styles
import styles from '../Styles/EntitesStyles/ThumbsLargeStyles'
import images from '../../Themes/Images'

const ThumbsLarge = ({ title, profileIcons }) => {
  const profileIconElement = profileIcons.map((profileIcon, index) => (
    <Image source={{uri:profileIcon}} key={index} style={styles.profileIcon} />
  ))
  return (
    <View style={styles.cardBox}>
      <View style={styles.profileIconBox}>{profileIconElement}</View>

      <View style={styles.textOutline}>
        <Text style={styles.textNameFont}>{title}</Text>
      </View>
    </View>
  )
}

ThumbsLarge.propTypes = {
  title: PropTypes.string,
  profileIcons: PropTypes.array
}

ThumbsLarge.defaultProps = {
  title: 'Default',
  profileIcons: [images.EntitiesImage0]
}

export default ThumbsLarge
