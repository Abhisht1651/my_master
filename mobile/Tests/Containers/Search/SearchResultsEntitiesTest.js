// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import TestFile from '../../../App/Containers/Search/SearchResultsEntities'
import Entities from '../../../App/Components/Entities/CardType1'

describe('Search Results Entities Screen', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<TestFile />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Entities Component', () => {
    const wrapper = shallow(<TestFile />)
    expect(wrapper.find(Entities)).toHaveLength(8)
  })
})
