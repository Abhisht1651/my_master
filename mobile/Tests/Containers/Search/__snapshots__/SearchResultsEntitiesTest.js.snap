// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results Entities Screen should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
  >
    <View
      style={
        Object {
          "marginVertical": 20,
          "width": 650,
        }
      }
    >
      <View
        key="0"
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <CardType1
          count={52}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          title="India"
        />
      </View>
      <View
        key="1"
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <CardType1
          count={52}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          title="United States of America"
        />
      </View>
      <View
        key="2"
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <CardType1
          count={52}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          title="India"
        />
      </View>
      <View
        key="3"
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <CardType1
          count={52}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          title="United States of America"
        />
      </View>
      <View
        key="4"
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <CardType1
          count={52}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          title="India"
        />
      </View>
      <View
        key="5"
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <CardType1
          count={52}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          title="United States of America"
        />
      </View>
      <View
        key="6"
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <CardType1
          count={52}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          title="India"
        />
      </View>
      <View
        key="7"
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <CardType1
          count={52}
          image1={
            Object {
              "default": "",
            }
          }
          onPressFollow={[Function]}
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          title="United States of America"
        />
      </View>
    </View>
  </ScrollViewMock>
</View>
`;
